src/types/platforms-types/heds-collab.ts

```diff
@@ -0,0 +1,142 @@
+    'LOPHIILE FT. TYLER JAY': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2FLophiile.jpg?alt=media&token=538507ad-08d7-499d-a7af-f689377f3349',
+    'CARRTOONS': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Fcarrtoons.jpg?alt=media&token=e6e5311c-590c-4a6e-b82d-3b3cc2bda207',
+    'ROMDERFUL': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2FROMDERFUL.jpg?alt=media&token=697ee0dc-a8ed-4a81-bc6b-ac31f9db22d0',
+    'AUTUMN KEYS FT AYOTEMI': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Fautumn%20keys.jpeg?alt=media&token=6cec0f08-839e-4427-b875-178a8ad5c788',
+    'OSHI': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Foshi.png?alt=media&token=c1282dae-5de3-44cd-91dc-d38c75ee4b07',
+    'DECAP': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Fdecap.jpg?alt=media&token=0788736c-6b9b-43c2-becd-be72b5578dbf',
+    'CHROMONICCI': `https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Fchromonicci.jpg?alt=media&token=cd5f28eb-350a-4c7f-a353-bf3894759bc7`,
+    'XANDER': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Fxander.jpeg?alt=media&token=57173553-96db-49a4-b739-f7284161f48e',
+    'CAPSHUN & CORDEROYBOI': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Fcapshun-4.jpeg?alt=media&token=37b17134-0b3f-43f6-89d7-456ff5925a92',
+    'PAUL MOND': 'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/tapes%2FcollabTAPES%2FGood%20Society%2FImages%2Fartists%2Fpaulmond.jpg?alt=media&token=bdb72778-07a7-43c6-a4aa-580b5fa073b3'
+  }
+
+  const track = avatarByArtist[artist.toUpperCase()];
+
+  if (!track){
+    throw `Can't map artist to avatar`;
+  }
+
+  return track
+
+}
+
+function getArtistId(artist: string){
+  const idByArtist: any = {
+    'OSHI': 'ethereum/0x4d18f8f2ae19f1e166c97793cceeb70680a2b6d2',
+    'CAPSHUN & CORDEROYBOI': 'ethereum/0xaa86ff6eb0ac77d46de48e955402cc3435c7ab8f',
+  }
+
+  const id = idByArtist[artist.toUpperCase()];
+
+  if (id){
+    return id
+  }
+
+  return artist
+
+}
+
+const mapTrack: MapTrack = (
+  nft: NFT,
+  apiTrack: any,
+  contract?: NftFactory,
+  trackId?: string,
+): ProcessedTrack => {
+  if (!contract) {
+    throw new Error(`Contract missing for mapTrack for nft ${nft.id}`)
+  }
+
+  const artist = getTrait(nft, 'Artist');
+
+  const track: Partial<ProcessedTrack> = {
```

Comment:
maybe we can minimize this code by importing a base mapTrack from another type and just extending it with the special fields?

Reply:
cool looks good - not really that impactful but interesting to experiment with how we may be able to compose mappers in a more powerful way with higher order functions like this in future